flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java

```diff
@@ -745,20 +749,44 @@ public void visit(ListConstructorExpressionNode listCtrExprNode) {
     }
 
     private void handleConstructorExpressionNode(ExpressionNode constructorExprNode, String typeDescription) {
-        SyntaxKind kind = constructorExprNode.parent().kind();
+        NonTerminalNode parent = constructorExprNode.parent();
+        SyntaxKind kind = parent.kind();
         if (kind == SyntaxKind.LOCAL_VAR_DECL || kind == SyntaxKind.MODULE_VAR_DECL) {
+            Optional<Symbol> parentSymbol = semanticModel.symbol(parent);
+            if (parentSymbol.isPresent() && isVarTypeJson(parentSymbol.get())) {
+                startNode(FlowNode.Kind.JSON_PAYLOAD)
+                        .metadata()
+                        .description(JSONPayload.DESCRIPTION)
+                        .stepOut()
+                        .properties().expression(constructorExprNode);
+                return;
+            }
```

Comment:
Seems like this is common for both conditions. Better to extract this out and have this as the first if condition.